pkg/registry/apis/provisioning/repository/changes.go

```diff
@@ -28,28 +28,26 @@ func Changes(source []FileTreeEntry, target *provisioning.ResourceList) ([]FileC
 		if ok {
 			if check.Hash != file.Hash {
 				changes = append(changes, FileChange{
-					Action:      FileActionUpdated,
-					Path:        file.Path,
-					Ref:         file.Hash,
-					PreviousRef: check.Hash,
+					Action: FileActionUpdated,
+					Path:   check.Path,
+					DB:     check,
```

Comment:
I am not sure (or don't remember) when did this file go into our prototype branch, but I have the same comment about this one. I think this should not reside in repository package. Maybe it's even a good moment to create a package for synchronization only. It will probably be one of the heaviest when it comes to business logic.